x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/RequestInfo.java

```diff
@@ -35,14 +35,25 @@
     
     private Mode mode;
     private String clientId;
-    
+    private Version version;
+
     public RequestInfo(Mode mode) {
-        this(mode, null);
+        this(mode, null, null);
     }
     
     public RequestInfo(Mode mode, String clientId) {
+        this(mode, clientId, null);
+    }
+
+    public RequestInfo(Mode mode, String clientId, String version) {
         mode(mode);
         clientId(clientId);
+        version(version);
+    }
+
+    public RequestInfo(Mode mode, Version version) {
```

Comment:
It seems this constructor is used everywhere using `Version.CURRENT` - maybe it makes sense to automatically initialize version when Mode is passed instead of setting it as `null`.
As it stands the constructor initializes Version twice.

Reply:
> As it stands the constructor initializes Version twice.

Addressed this.
But auto-initialising along with Mode is not possible, since the `CURRENT` Version value is only available in `client`, but not in `proto`.